### DISCUSSION TITLE
How can i reach user ip on +page.server.js or/and +layout.server.js

### DISCUSSION THREAD

**Discussion by Necoo33** [sveltejs/kit#11480] [Q&A] [answered]
Hello, i'm an open source contributor who developed several liblaries especially for node.js. I want to be able to reach user ip on server side and i don't know how to do that. So how can i reach users ip address on page.server.js and layout.server.js? 

An also is there any middleware mechanism in this framework which i could reach both request and response objects?

**Comment by Angus-Paillaugue** [comment, marked as answer]
Hi, if I am not mistaking, you can use the [`e.getClientAddress()`](https://kit.svelte.dev/docs/adapter-node#environment-variables-addressheader-and-xffdepth) function on a server function that returns the client ip adress : 
```
export async function load(e) {
  console.log(e.getClientAddress());
}
```
